pymongo/mongo_client.py

```diff
@@ -2393,12 +2384,15 @@ def _write(self) -> T:
         try:
             max_wire_version = 0
             self._server = self._get_server()
-            supports_session = (
-                self._session is not None and self._server.description.retryable_writes_supported
-            )
             with self._client._checkout(self._server, self._session) as conn:
                 max_wire_version = conn.max_wire_version
-                if self._retryable and not supports_session:
+                sessions_supported = (
+                    self._retryable
+                    and self._server.description.retryable_writes_supported
```

Comment:
Can we remove this line?

Reply:
And just check `conn.supports_sessions` instead?

Reply:
We can remove two of the variables in the check, the remaining two are required for tests to pass.